cmd/principal/main.go

```diff
@@ -275,6 +280,9 @@ func NewPrincipalRunCommand() *cobra.Command {
 	command.Flags().BoolVar(&enableResourceProxy, "enable-resource-proxy",
 		env.BoolWithDefault("ARGOCD_PRINCIPAL_ENABLE_RESOURCE_PROXY", true),
 		"Whether to enable the resource proxy")
+	command.Flags().DurationVar(&waitDurationForAgentPing, "wait-duration-for-agent-ping",
```

Comment:
With the explanation that you gave me for the purpose of this switch, I think the phrase "wait duration" might be a bit misleading here. To me, this phrase implies that the principal waits some time for the agent ping, when in reality it enforces agent pings to not being sent at a shorter interval than the value given. Good command line option names should be more or less self-explaining.

So for this, `enforce-min-keepalive` or `keepalive-min-interval` would be two options that come to my mind. I'm open for suggestions of other names.